source/adios2/ADIOSTypes.h

```diff
@@ -137,6 +114,35 @@ struct TypeInfo<T, typename std::enable_if<std::is_same<
     using ValueType = typename T::value_type;
 };
 
+const size_t UnknownDim = 0;
+const size_t JoinedDim = SIZE_MAX - 1;
```

Comment:
I just checked with GCC4.8, which is as far back as we will support, and it's constexpr